Update accessibility contrast test coverage

ankur2136 opened the pull request:
Update the accessibility test to cover all Elements with same semantic label to get better coverage on contrast test.

#109465 Improve flutter contrast test coverage


## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

ankur2136 committed:
Update accessibility.dart

Update the accessbility test to cover all Elements with same semantic label to get better coverage on contrast test.

ankur2136 committed:
Update accessibility.dart

Update the accessbility test to cover all Elements with same semantic label to get better coverage on contrast test.

ankur2136 committed:
variable fix.

Nit fix.

ankur2136 committed:
variable fix.

Nit fix.

yusuf-goog committed:
Updating expiring cirrus key. (#109461)

engine-flutter-autoroll committed:
Roll Flutter Engine from de118a55ade0 to c354e0e04cea (2 revisions) (#109462)

ankur2136 committed:
Adding test

ankur2136 committed:
Adding test

ankur2136 committed:
test fix.

ankur2136 committed:
test fix.

ankur2136 committed:
nit fix.

ankur2136 committed:
nit fix.

ankur2136 committed:
Nit fix

ankur2136 committed:
Nit fix

engine-flutter-autoroll committed:
Roll Flutter Engine from c354e0e04cea to d1b18b5d6811 (1 revision) (#109466)

goderbauer committed:
Keep `dirty` manipulations private to `Element` base class (#109401)

jmagman committed:
Remove deprecated Ruby File.exists? in helper script (#109428)

gspencergoog committed:
Revert "Keep `dirty` manipulations private to `Element` base class (#109401)" (#109481)

Landing on red to green the build.

engine-flutter-autoroll committed:
Roll Flutter Engine from d1b18b5d6811 to a2f3bd58ce73 (1 revision) (#109472)

Renzo-Olivares committed:
Single tap on the previous selection should toggle the toolbar on iOS… (#108913)

engine-flutter-autoroll committed:
Roll Flutter Engine from a2f3bd58ce73 to fb19700742c3 (6 revisions) (#109482)

engine-flutter-autoroll committed:
Roll Flutter Engine from fb19700742c3 to 2180e62f6c7e (1 revision) (#109483)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2180e62f6c7e to cd1a1cb757f5 (1 revision) (#109488)

engine-flutter-autoroll committed:
Roll Flutter Engine from cd1a1cb757f5 to 5cc5c5f04f0a (1 revision) (#109490)

engine-flutter-autoroll committed:
Roll Flutter Engine from 5cc5c5f04f0a to 18dca3de7496 (2 revisions) (#109493)

engine-flutter-autoroll committed:
Roll Flutter Engine from 18dca3de7496 to 83771593cb55 (1 revision) (#109505)

Hixie committed:
Cleaner test.dart output. (#109206)

engine-flutter-autoroll committed:
Roll Flutter Engine from 83771593cb55 to f49a617535af (1 revision) (#109509)

godofredoc committed:
Update README.md (#109506)

limonadev committed:
Add onOpened callback to PopupMenuButton (#103753)

engine-flutter-autoroll committed:
Roll Flutter Engine from f49a617535af to cc9f4c63b7e5 (1 revision) (#109518)

engine-flutter-autoroll committed:
Roll Flutter Engine from cc9f4c63b7e5 to 4656c2f46cad (1 revision) (#109524)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4656c2f46cad to 16fb19e72f88 (1 revision) (#109526)

engine-flutter-autoroll committed:
Roll Flutter Engine from 16fb19e72f88 to 9c3c233e2639 (1 revision) (#109532)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9c3c233e2639 to f0c3829e90ce (1 revision) (#109536)

engine-flutter-autoroll committed:
Roll Plugins from 9fb7654f3aa2 to 094f1c3ad94a (7 revisions) (#109543)

engine-flutter-autoroll committed:
Roll Flutter Engine from f0c3829e90ce to 8c019cdd446f (2 revisions) (#109548)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8c019cdd446f to 198b0051a5a2 (1 revision) (#109552)

engine-flutter-autoroll committed:
Roll Flutter Engine from 198b0051a5a2 to 33fccf564973 (1 revision) (#109557)

goderbauer committed:
Reland "Keep `dirty` manipulations private to `Element` base class (#109401)" (#109562)

engine-flutter-autoroll committed:
Roll Flutter Engine from 33fccf564973 to b4cbebe3887e (3 revisions) (#109564)

bernaferrari committed:
Refactor StrokeAlign to allow double values. (#108339)

yaakovschectman committed:
Add Compilation DeviceLab test for Windows (#109540)

* Add Windows compilation test (hello_world_windows__compile)

* Add devicelab target to .ci.yaml

* Bringup on new target

* Add license to generated files

* Indicate test ownership

* Delete asset from repo

* Add Windows compilation test (hello_world_windows__compile)

* Add devicelab target to .ci.yaml

* Bringup on new target

* Add license to generated files

* Indicate test ownership

* Delete icon

* Use path.basename

Re triggering tests

* Update test owner to desktop team

engine-flutter-autoroll committed:
Roll Flutter Engine from b4cbebe3887e to bf9f07b7b573 (1 revision) (#109566)

engine-flutter-autoroll committed:
Roll Flutter Engine from bf9f07b7b573 to 2c3efa0beb24 (1 revision) (#109570)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2c3efa0beb24 to 90477c21aab5 (2 revisions) (#109575)

GaryQian committed:
Update `flutter.gradle` AGP to 7.2.0 and bump default NDK version (#109211)

engine-flutter-autoroll committed:
Roll Flutter Engine from 90477c21aab5 to de2640e525d9 (2 revisions) (#109582)

engine-flutter-autoroll committed:
Roll Flutter Engine from de2640e525d9 to 90d845191153 (2 revisions) (#109584)

xu-baolin committed:
Fix Text selection handler disappear when dragged to new words (#108808)

engine-flutter-autoroll committed:
Roll Flutter Engine from 90d845191153 to bbe0038e2a36 (1 revision) (#109586)

engine-flutter-autoroll committed:
Roll Flutter Engine from bbe0038e2a36 to 280e9a7becad (1 revision) (#109587)

Commit:
Revert "Refactor StrokeAlign to allow double values." (#109591)

engine-flutter-autoroll committed:
Roll Flutter Engine from 280e9a7becad to 8a1ab4692ae0 (1 revision) (#109597)

TahaTesser committed:
Fix `CupertinoListTile``s onTap with delay throws exception (#109038)

maheshj01 committed:
Pass fit property to RenderIndexedStack (#109295)

engine-flutter-autoroll committed:
Roll Flutter Engine from 8a1ab4692ae0 to db741c85ad09 (1 revision) (#109611)

engine-flutter-autoroll committed:
Roll Plugins from 094f1c3ad94a to d18079b41616 (9 revisions) (#109615)

fzyzcjy committed:
Fix that RenderEditable (TextField) ignores offset in painting, making text selections shifted when offset is nonzero (#109287)

Small fix in text layout math, unlikely to affect most normal usages.

chunhtai reviewed (approved):
LGTM

chunhtai commented:
CC @goderbauer for secondary review

camsim99 committed:
Add Spellcheck to EditableText (Android) (#109334)

goderbauer commented on packages/flutter_test/lib/src/accessibility.dart:
```diff
@@ -331,50 +331,52 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+  }
 
+  Future<Evaluation> _evaluateElement(
+      SemanticsNode node,
+      Element element,
+      WidgetTester tester,
+      ui.Image image,
+      ByteData byteData,
+      ) async {
```
nit: indentation:

```suggestion
    SemanticsNode node,
    Element element,
    WidgetTester tester,
    ui.Image image,
    ByteData byteData,
  ) async {
```

ankur2136 replied:
Updated.

goderbauer commented on packages/flutter_test/lib/src/accessibility.dart:
```diff
@@ -331,50 +331,52 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+      ui.Image image,
+      ByteData byteData,
+      ) async {
+    const Evaluation result = Evaluation.pass();
```
What is this extra const variable for? Can you just use `const Evaluation.pass()` in place of `result` below to clear up the code?

ankur2136 replied:
Updated.

goderbauer reviewed (approved):
LGTM after comments are resolved.

ankur2136 committed:
Comment fixes.

ankur2136 committed:
Comment fixes.

goderbauer commented on packages/flutter_test/lib/src/accessibility.dart:
```diff
@@ -331,61 +331,62 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
+    WidgetTester tester,
+    ui.Image image,
+    ByteData byteData,
+    ) async {
```
unindent this line by two spaces.

ankur2136 replied:
Done

ankur2136 committed:
unindent

ankur2136 committed:
unindent

ankur2136 committed:
unindent

ankur2136 committed:
unindent

Commit:
Revert "Add Spellcheck to EditableText (Android) (#109334)" (#109631)

This reverts commit ff9fe35682e10d1067d918770048d28a1f462e0f.

engine-flutter-autoroll committed:
Roll Flutter Engine from db741c85ad09 to 688624b057db (1 revision) (#109617)

jonahwilliams commented:
Won't this pick up text that has been excluded from semantics?

olof-dev committed:
Change "Unicode scalar values" to "Unicode grapheme clusters" in maxLength docs (#106275)

ankur2136 commented:
> Won't this pick up text that has been excluded from semantics?

Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.

jonahwilliams commented:
Also, if we're going to look at every text element anyway - then why bother looking at the semantics nodes at all. Shouldn't we just traverse the element tree to gather every text element?

engine-flutter-autoroll committed:
Roll Flutter Engine from 688624b057db to 69ee0c7094f3 (12 revisions) (#109634)

goderbauer reviewed (changes requested):
>> Won't this pick up text that has been excluded from semantics?
>
> Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.

Can you add a test case to verify that this isn't happening, please?

goderbauer commented:
@jonahwilliams has a good point regarding element traversal...

gspencergoog committed:
Some changes needed to PlatformMenuBar before the MenuBar implementation change lands. (#109556)

engine-flutter-autoroll committed:
Roll Flutter Engine from 69ee0c7094f3 to eb7428a51ddd (3 revisions) (#109639)

engine-flutter-autoroll committed:
Roll Flutter Engine from eb7428a51ddd to 028a5bcebfab (3 revisions) (#109647)

xu-baolin committed:
Scrollbar respect the NeverScrollableScrollPhysics physics (#109609)

LongCatIsLooong committed:
Fix a type casting error in text_input.dart (#109635)

engine-flutter-autoroll committed:
Roll Flutter Engine from 028a5bcebfab to 246f24589107 (1 revision) (#109651)

engine-flutter-autoroll committed:
Roll Flutter Engine from 246f24589107 to cd35ae3706a4 (3 revisions) (#109653)

engine-flutter-autoroll committed:
Roll Flutter Engine from cd35ae3706a4 to 2e6d4dc5b513 (1 revision) (#109657)

engine-flutter-autoroll committed:
Roll Flutter Engine from 2e6d4dc5b513 to 4024ff17ac5f (3 revisions) (#109660)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4024ff17ac5f to fa91de4b2072 (1 revision) (#109662)

engine-flutter-autoroll committed:
Roll Flutter Engine from fa91de4b2072 to e480e68d34a7 (1 revision) (#109665)

engine-flutter-autoroll committed:
Roll Flutter Engine from e480e68d34a7 to 899cc167aa7c (1 revision) (#109667)

engine-flutter-autoroll committed:
Roll Flutter Engine from 899cc167aa7c to b9ba9ce9b510 (2 revisions) (#109670)

yaakovschectman committed:
Bring up new Windows start-up tests (#109618)

* Add Windows startup test

* Add new tests for Windows desktop

* Bring up complex layout tests for Win Desktop

* Bring up flutter_view startup test for Windows

* Add Platform View startup test for Windows

* Fix typo in .ci.yaml

* Make basename variable final

* Link to bug in .ci.yaml

engine-flutter-autoroll committed:
Roll Flutter Engine from b9ba9ce9b510 to 295776ac1ff3 (1 revision) (#109681)

engine-flutter-autoroll committed:
Roll Flutter Engine from 295776ac1ff3 to 5c141850e427 (1 revision) (#109686)

engine-flutter-autoroll committed:
Roll Plugins from d18079b41616 to 0d3d520952c1 (11 revisions) (#109688)

camsim99 committed:
Re-land Add Spell Check to EditableText  (#109643)

craiglabenz committed:
[docs][FWW] DropdownButton, ScaffoldMessenger, and StatefulBuilder links (#100316)

* added dropdown and scaffoldmessenger FWW links

* added StatefulBuilder wotw link

* added Autocomplete FWW link

* removed accidental auto-formatting

engine-flutter-autoroll committed:
Roll Flutter Engine from 5c141850e427 to fa316fccb372 (1 revision) (#109691)

engine-flutter-autoroll committed:
Roll Flutter Engine from fa316fccb372 to df186d820011 (1 revision) (#109699)

ankur2136 committed:
Additional test for multiple nodes with same semantic label.

Add a test with multiple nodes with same semantic label. but one of the label is excluded from semantic tree and has failing contrast (white on white) but this should pass a11y contrast test.

ankur2136 committed:
Additional test for multiple nodes with same semantic label.

Add a test with multiple nodes with same semantic label. but one of the label is excluded from semantic tree and has failing contrast (white on white) but this should pass a11y contrast test.

ankur2136 commented:
> > > Won't this pick up text that has been excluded from semantics?
> > 
> > 
> > Hasn't happened in any test in GP3. The fix has identified 470 tests however where we needed to manually handle failures.
> 
> Can you add a test case to verify that this isn't happening, please?

Added another test with a contrast failing Text wrapped in ExcludeSemantics and a duplicate Text with contrast within guidelines.

fluttergithubbot committed:
Marks Windows_android channels_integration_test_win to be flaky (#109695)

engine-flutter-autoroll committed:
Roll Flutter Engine from df186d820011 to ba62c6510edc (2 revisions) (#109704)

engine-flutter-autoroll committed:
Roll Flutter Engine from ba62c6510edc to 3df3a6e8eba5 (2 revisions) (#109710)

bernaferrari committed:
Fix AppBar centerTitle position with actions. (#106256)

bernaferrari committed:
Improve ShapeDecoration performance. (#108648)

pdblasi-google committed:
109638: Windows framework_tests_misc is 2.06% flaky (#109640)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3df3a6e8eba5 to 6f8c99debd14 (2 revisions) (#109713)

aam committed:
Introduce stubbed `exclusive` parameter to `File.create`-overridden method (#109646)

* Introduce stubbed `exclusive` parameter to `File.create`-overridden method.

This is to soften landing of a breaking change https://github.com/dart-lang/sdk/issues/49647

gspencergoog committed:
Some misc changes needed for MenuBar implementation. (#109555)

engine-flutter-autoroll committed:
Roll Flutter Engine from 6f8c99debd14 to a91226d88602 (3 revisions) (#109716)

camsim99 committed:
Remove scheduler (#109714)

engine-flutter-autoroll committed:
Roll Flutter Engine from a91226d88602 to 4b8d1fad35cd (2 revisions) (#109719)

keyonghan committed:
Update test owners (#109717)

engine-flutter-autoroll committed:
Roll Flutter Engine from 4b8d1fad35cd to e8d5233a81b6 (1 revision) (#109725)

moffatman committed:
Don't disable pointer interaction during trackpad scroll (#106890)

engine-flutter-autoroll committed:
Roll Flutter Engine from e8d5233a81b6 to fbbd8c6bab64 (1 revision) (#109727)

engine-flutter-autoroll committed:
Roll Flutter Engine from fbbd8c6bab64 to 32d8c521da4b (3 revisions) (#109744)

engine-flutter-autoroll committed:
Roll Flutter Engine from 32d8c521da4b to 654f8029aad4 (1 revision) (#109748)

hellohuanlin committed:
[platform_view]terminate the app in tear down for platform view xcuitest (#109720)

engine-flutter-autoroll committed:
Roll Flutter Engine from 654f8029aad4 to a77ead45a80a (1 revision) (#109752)

engine-flutter-autoroll committed:
Roll Flutter Engine from a77ead45a80a to 3d85fe7ee454 (1 revision) (#109760)

engine-flutter-autoroll committed:
Roll Flutter Engine from 3d85fe7ee454 to 9807e5ce95c7 (1 revision) (#109763)

engine-flutter-autoroll committed:
Roll Flutter Engine from 9807e5ce95c7 to 1311fea574d6 (2 revisions) (#109765)

TahaTesser committed:
Add an example for how to hide default scrollbar on desktop platform. (#108542)

yaakovschectman committed:
Add windows implementation to platform_view example (#109715)

* Add Windows platform view example

* Add new tests for Windows desktop

* Add Windows platform view

* Use const constructors (linux analyze)\

* Fix spelling/format

* Clean up code

* Remove default case

engine-flutter-autoroll committed:
Roll Plugins from 0d3d520952c1 to 4395e3195be7 (5 revisions) (#109777)

jensjoha committed:
[flutter_tools] Make flutter test -v print timing of different phases (#108864)

goderbauer commented:
You probably have to rebase this with the latest master to get tests to pass again.

QuncCccccc committed:
Reland: Set IconButton.visualDensity default to VisualDensity.standard (#109432)

* Revert "Revert "Set `IconButton.visualDensity` default to `VisualDensity.standard` (#109349)" (#109421)"

Co-authored-by: Qun Cheng <quncheng@google.com>

ankur2136 committed:
Merge branch 'a11y' of https://github.com/ankur2136/flutter into a11y

ankur2136 commented:
Closed this since it caused a bad merge with master and somehow went ahead of flutter/master. 

Created a new PR in https://github.com/flutter/flutter/pull/109784
